### PR TITLE
Added date_trunc time/date scalar function

### PR DESCRIPTION
**Pull request by lorycloutier**
FB-1961
Added function and test coverage.

### CODE REVIEW

File: sql3/planner/inbuiltfunctionsdate.go [hunk 1]
```diff
@@ -466,6 +488,76 @@ func (n *callPlanExpression) EvaluateDatetimeAdd(currentRow []interface{}) (inte
...
+	if !intervalOk {
+		return nil, sql3.NewErrInternalf("unable to convert value")
+	}
+
+	switch strings.ToUpper(interval) {
+	case intervalYear:
+		return fmt.Sprint(date.Year()), nil
+
+	case intervalMonth:
+		return fmt.Sprintf("%d-%02d", date.Year(), date.Month()), nil
```
Comment (rachithrr) [inline sql3/planner/inbuiltfunctionsdate.go:534]: I'm curious, Did you try time.Format() ? 
```
return date.Format("2006-01"), nil
return date.Format("2006-01-02"), nil
return date.Format("2006-01-02T15"), nil
.....
```

**Review by rachithrr — COMMENTED** [review]
Looks good. 
1 question about time.Format()

File: sql3/planner/inbuiltfunctionsdate.go (hunk 1 above)
Comment (lorycloutier) [reply sql3/planner/inbuiltfunctionsdate.go]: Missed seeing that originally, converted to using it.

**Review by rachithrr — APPROVED** [review]
looks good to me

**Pull request merged**